symmetry_values_test.go

```diff
@@ -0,0 +1,387 @@
+package gocql
+
+import (
+	"github.com/google/uuid"
+	"gopkg.in/inf.v0"
+	"math"
+	"math/big"
+	"net"
+	"reflect"
+	"strconv"
+	"testing"
+	"time"
+)
+
+const testProto = 2
+
+func TestSymmetry(t *testing.T) {
```

Comment:
There is two problems with this implementations:
1. It does not actually check marshal correctness, meaning that it does not check that certain structure end up marshaled into certain data
2. It is so hard to understand what exactly is tested that you need to debug it to see it.

To address that please redo tests in the following format, don't replicate it `100%`, use it just as point of reference:
```
type marshalTypeTestCase struct {
	name          string
	cqlType       TypeInfo
	value         interface{}
	expected      []byte
	skipUnmarshal string
}

var marshalTestCases = []marshalTypeTestCase{}

func TestMarshal(t *testing.T) {
	for id := range marshalTestCases {
		tcase := marshalTestCases[id]
		tcase.cqlType.Version()
		t.Run(fmt.Sprint("%s-%d-%s", tcase.cqlType.Type().String(), tcase.cqlType.Version(), tcase.name), func(t *testing.T) {
			data, err := Marshal(tcase.cqlType, tcase.value)
			if err != nil {
				t.Fatal("failed to marshal value", err)
			}
			if !reflect.DeepEqual(data, tcase.expected) {
				t.Fatalf("expected %v, got %v", tcase.expected, data)
			}

			gotUnmarshal := reflect.New(reflect.TypeOf(tcase.value))
			err = Unmarshal(tcase.cqlType, data, gotUnmarshal)
			if err != nil {
				t.Fatal("failed to unmarshal to the same type", err)
			}

			if tcase.skipUnmarshal != "" {
				t.Skip(tcase.skipUnmarshal)
				return
			}

			if !reflect.DeepEqual(tcase.value, gotUnmarshal) {
				t.Fatalf("expected %v, got %v", tcase.expected, data)
			}
		})
	}
}
```

Also please reconcile with what already exists in `marshal_test.go` and remove old duplicates.

Reply:
Fixed